app/services/git_adapter.rb

```diff
@@ -1,6 +1,21 @@
+require 'fileutils'
+
 class GitAdapter
-  def self.clone_repo(repo_url:, to_dir: dir)
-    ShellAdapter.exec(git_binary, 'clone', repo_url, to_dir)
+  # TODO: this is not thread/process safe and does not scale
```

Comment:
Is there anywhere better to put `TODOs` than the code? Perhaps in a tech debt board or something?

Reply:
I've created a ticket for this `FB-413` under the `tech debt` epic. It is a bigger issue as i think we should remove this entire class and concept and replace this mechanism with using the github http api which would resolve this problem.